references/embedding/sampler.py

```diff
@@ -0,0 +1,63 @@
+import torch
+from torch.utils.data.sampler import Sampler
+from collections import defaultdict
+import random
+
+
+def create_groups(groups, k):
+    """Bins sample indices with respect to groups, remove bins with less than k samples
+
+    Args:
+        groups (list[int]): where ith index stores ith sample's group id
+
+    Returns:
+        defaultdict[list]: Bins of sample indices, binned by group_idx
+    """
+    group_samples = defaultdict(list)
+    for sample_idx, group_idx in enumerate(groups):
+        group_samples[group_idx].append(sample_idx)
+
+    keys_to_remove = []
+    for key in group_samples:
+        if len(group_samples[key]) < k:
+            keys_to_remove.append(key)
+            continue
+
+    for key in keys_to_remove:
+        group_samples.pop(key)
+
+    return group_samples
+
+
+class PKSampler(Sampler):
```

Comment:
Can you add some documentation to this sampler?

Reply:
I described what the sampler does and explained what `p` and `k` meant in the `Args:` section.